### PR TITLE
LO-2261: Handle FHIR Datetimes

### PR DESCRIPTION
**Pull request by tdstein**
This change fixes the FHIR date time parsing and migrates from joda-time to the 'java.time' package.

### CODE REVIEW

**Review by david-wb — APPROVED** [review]
Nice update. I don't see any problems.

**Review by dmmiller612 — APPROVED** [review]

File: src/main/scala/com/lifeomic/fhirlib/v3/Deserializer.scala [hunk 1]
```diff
@@ -1,26 +1,25 @@
 package com.lifeomic.fhirlib.v3
 
+import java.time.format.DateTimeFormatter
+import java.time.{LocalDateTime}
+
 import com.lifeomic.fhirlib.v3.datatypes._
 import com.lifeomic.fhirlib.v3.resources.{Resource, _}
-import org.joda.time.format.ISODateTimeFormat
```
Comment (wavetylor) [inline src/main/scala/com/lifeomic/fhirlib/v3/Deserializer.scala]: If we've removed all joda usages, can we remove it from the POM?

Comment (tdstein) [reply src/main/scala/com/lifeomic/fhirlib/v3/Deserializer.scala]: Yes - it's already gone. We actually never explicitly imported it. It was grabbing the version from json4s.

**Review by dmmiller612 — APPROVED** [review]

**Review by david-wb — APPROVED** [review]

**Review by david-wb — APPROVED** [review]

**Review by mjtieman — APPROVED** [review]
LGTM, nice work

**Review by wavetylor — APPROVED** [review]

**Pull request merged**